plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java

```diff
@@ -80,6 +80,7 @@ public void search(SearchRequest request, ActionListener<SearchResponse> actionL
         User user = RestActionUtils.getUserContext(client);
         ActionListener<SearchResponse> listener = wrapRestActionListener(actionListener, "Fail to search model version");
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            ActionListener<SearchResponse> wrappedListener = ActionListener.runBefore(listener, () -> context.restore());
```

Comment:
I assume we can't provide this listener as generic from utils because context doesn't exist?

Reply:
yes